numba/core/ir_utils.py

```diff
@@ -2215,15 +2210,16 @@ def legalize_single_scope(blocks):
     return len({blk.scope for blk in blocks.values()}) == 1
 
 
-def check_and_legalize_ir(func_ir):
+def check_and_legalize_ir(func_ir, flags: "numba.core.compiler.Flags"):
     """
     This checks that the IR presented is legal
     """
     enforce_no_phis(func_ir)
     enforce_no_dels(func_ir)
     # postprocess and emit ir.Dels
     post_proc = postproc.PostProcessor(func_ir)
-    post_proc.run(True, extend_lifetimes=config.EXTEND_VARIABLE_LIFETIMES)
+    ext_life = flags.dbg_extend_lifetimes or config.EXTEND_VARIABLE_LIFETIMES
```

Comment:
IIRC the environment variables are usually overridden by in-source definition?

Reply:
good point